erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -121,9 +121,9 @@ def get_so_items(self):
 			item_condition = ' and so_item.item_code = {0}'.format(frappe.db.escape(self.item_code))
 
 		items = frappe.db.sql("""select distinct parent, item_code, warehouse,
-			(qty - work_order_qty) * conversion_factor as pending_qty, description, name
+			(qty - work_order_qty - delivered_qty) * conversion_factor as pending_qty, description, name
```

Comment:
This becomes problematic when `delivered_qty` is also manufactured via a work order.

E.g. assume everything is for a single item

SO - 100 Qty
DN - 10 Qty
WO - 50 Qty
DN - 50 Qty

2nd WO = 100 - 50 - 60 = -10 qty (which doesn't get added in table)

So in the case of multi-stage production plans. This will cause double subtraction of manufactured quantity.

Reply:
Can you also change this and subtract produced quantity?

Reply:
This _might_ work:

Pending qty = required qty - (WO qty - Produced qty) - delivered qty

Reply:
It seems to me this is a symptom of multiple accounting on the Sales Order Item.

Are you saying that all of these quantities are additive only, so nothing is subtracted (unless cancelled)? Which means they will always add up to more than the ordered quantity?

Seems to me this needs a re-think. If a work order is complete, why is it still a work order quantity and not transferred to Produced Quantity?
Shouldn't a plan take into account the Planned Quantity?
No idea what Projected, Actual and Ordered Quantities are.

![Screen Shot 2021-07-31 at 01 44 36](https://user-images.githubusercontent.com/64409021/127723438-a7f60503-aeab-4002-9000-dfff40441a17.png)

Reply:
Logically this is what I'd expect:
pending_qty = qty - delivered_qty - actual_qty - ordered_qty - planned_qty - work_order_qty - produced_qty + returned_qty

Expected transfer of quantities (debit one, credit another):
actual -> delivered
ordered -> actual -> delivered
planned -> work_order -> produced -> delivered
returned is effectively an addition to the original qty that must be re-delivered

actual_qty = available_stock_qty ?
ordered_qty = purchase_ordered_qty ?

Whether attempting to account for everything against the sales order item is a good idea or not is another question.
It's certainly brave.